### PR TITLE
New control algorithm

### PR DESCRIPTION
**Pull request by nagasai-iitr**
The previous commit contains the algorithm for controlling the robots' movements in terms of speed and direction but here I have updated the file so as to control the robot in such a way one can control a car in a normal game. 
Even though we here are restricting the complete use of 
```
differentialrobot_proxy.setSpeedBase(adv, rot)
```
but since ultimately the use of this component is in real life in which people's comfort of using the gear or component is the first priority of making the component 

NOTE : Here the values of speed are discrete but we can omit the code evenmore by adding one more button for changing the discrete values

### CODE REVIEW

**Comment by nagasai-iitr** [comment]
Please look into this

**Comment by orensbruli** [comment]
Hello @nagasai-iitr ,

Thanks for your contribution.
I tell you my doubts about this change. This introduces a library dependency to pygame that is only used in this component. This component is mostly used as an example component. The less the people have to install to make it work, the better.

A different approximation could be to make the import of pygame in a try-except. If it's imported, the new algorithm could be called, if not, the old one. Both algorithms would be moved to different methods and be called from compute...

What do you think?
Best.

**Comment by nagasai-iitr** [comment]
Yeah I will change it by including try-except method 

Thankyou

**Comment by nagasai-iitr** [comment]
@orensbruli I have changed the file for the corresponding needs

**Comment by nagasai-iitr** [comment]
Please look into this.

File: components/hardware/external_control/keyboardrobotcontroller/src/specificworker.py [hunk 1]
```diff
@@ -56,12 +56,20 @@ class SpecificWorker(GenericWorker):
...
+			self.timer.timeout.connect(self.compute1)
+			self.Period = 1
+			pygame.init()
+			pygame.display.set_mode((1,1), pygame.NOFRAME)
+			self.timer.start(self.Period)
+			screen.addstr(0,0,'Connected to robot. Use arrows (or) ASWD to control direction,Shift to speed, space bar to stop ans ''q'' to exit')
+		except ImportError as e:
+			self.timer.timeout.connect(self.compute2)
+			self.Period = 1
+			self.timer.start(self.Period)
```
Comment (orensbruli) [inline components/hardware/external_control/keyboardrobotcontroller/src/specificworker.py]: Repeated in the try and the except.

File: components/hardware/external_control/keyboardrobotcontroller/src/specificworker.py [hunk 2]
```diff
@@ -72,7 +80,7 @@ def setParams(self, params):
 	tt2=2
 
 	@QtCore.Slot()
-	def compute(self):
+	def compute1(self):
```
Comment (orensbruli) [inline components/hardware/external_control/keyboardrobotcontroller/src/specificworker.py]: more descriptive names

File: components/hardware/external_control/keyboardrobotcontroller/src/specificworker.py [hunk 3]
```diff
@@ -158,6 +166,54 @@ def compute(self):
 			traceback.print_exc()
 			print (e)
 		return True
+	
+	def compute2(self):
```
Comment (orensbruli) [inline components/hardware/external_control/keyboardrobotcontroller/src/specificworker.py]: more descriptive names

**Comment by nagasai-iitr** [comment]
@orensbruli  thanks for mentioning the above things to take care

**Comment by orensbruli** [comment]
@nagasai-iitr Can you take a look at the latest changes I made on your fork and test it. Set the pygame_flag to false to test the ncurses option.

**Comment by nagasai-iitr** [comment]
@orensbruli  Thank you for the above changes ,and the component is working perfectly alright even for ncurses and pygame

**Comment by orensbruli** [comment]
> @orensbruli Thank you for the above changes, and the component is working perfectly alright even for ncurses and pygame
Oh, really? Not sure.

**Comment by nagasai-iitr** [comment]
Here is the link of the folder containing video recording for using keyboardcontroller with pygame_compute method and ncurses_compute method and a document explaining the details for the video.
https://drive.google.com/drive/folders/1qPOPm66XuPtvpuDrcU-keaarVp8X-PCe?usp=sharing
Thank you

**Comment by orensbruli** [comment]
Thank you.

**Pull request merged**